File: algorithm_java/DAY3/boj_1874.java
package DAY3;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class boj_1874 {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static StringBuilder sb = new StringBuilder();

    public static void main(String[] args) throws IOException {
        try {
            int n = Integer.parseInt(br.readLine());
            Stack<Integer> stk = new Stack<>();

            int bp = 0;
            while (n > 0) {
                int k = Integer.parseInt(br.readLine());

                if (k > bp) {
                    for (int i = bp + 1; i <= k; i++) {
                        stk.push(i);
                        sb.append("+\n");
                    }
                    bp = k;
                }

                if (stk.peek() != k) {
                    System.out.println("NO");
                    return;
                }

                stk.pop();
                sb.append("-\n");

                n--;
            }

            System.out.println(sb);
        }
        finally {
            if (br != null) br.close();
        }
    }
}

File: algorithm_cpp/Project1/Algorithm_Session/boj_1874.cpp
#include <iostream>
#include <stack>

using namespace std;

void boj_1874() {
	
	int n;
	cin >> n;

	stack<int> stk;
	string ans = "";
	int bp = 0;

	while (n--) {
		int k;
		cin >> k;

		if (k > bp) {
			for (int i = bp + 1; i <= k; i++) {
				ans += "+\n";
				stk.push(i);
			}

			bp = k;
		}

		/*
		exit : 바로 C 프로그램 자체 프로세스 종료 (0 : 정상종료, 1 : 에러메세지 종료)
		return : 뒤 문장을 실행하며 종료
		*/

		if (stk.top() != k) {
			cout << "NO";
			exit(0);
		}

		ans += "-\n";
		stk.pop();
	}

	cout << ans;
}